jaxley/integrate.py

```diff
@@ -311,4 +315,5 @@ def _body_fun(state, externals):
         nested_lengths=checkpoint_lengths,
     )
     recs = jnp.concatenate([init_recording, recordings[:nsteps_to_return]], axis=0).T
+    # recs = recs[sort_inds, :]  # Sort recordings back to order that was set by user.
```

Comment:
This is a bit funky, when calling `df.groupby` you may be changing the order of the original `module.recordings`

This is meant to take care of that by putting things back to the original order, but uncommenting this line causes tests to fail so something in my logic is wrong

Reply:
nevermind! silly mistake, needed to use `argsort`